idea/src/org/jetbrains/kotlin/idea/inspections/MoveSuspiciousCallableReferenceIntoParenthesesInspection.kt

```diff
@@ -43,26 +45,36 @@ class MoveSuspiciousCallableReferenceIntoParenthesesInspection : AbstractKotlinI
                     }
                 }
                 holder.registerProblem(
-                        lambdaExpression,
-                        "Suspicious callable reference as the only lambda element",
-                        ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
-                        IntentionWrapper(MoveIntoParenthesesIntention(), lambdaExpression.containingFile)
+                    lambdaExpression,
+                    "Suspicious callable reference as the only lambda element",
+                    ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                    IntentionWrapper(MoveIntoParenthesesIntention(), lambdaExpression.containingFile)
                 )
 
             }
         })
     }
 
     class MoveIntoParenthesesIntention : ConvertLambdaToReferenceIntention(
-            "Move suspicious callable reference into parentheses '()'"
+        "Move suspicious callable reference into parentheses '()'"
     ) {
         override fun buildReferenceText(element: KtLambdaExpression): String? {
             val callableReferenceExpression =
-                    element.bodyExpression?.statements?.singleOrNull() as? KtCallableReferenceExpression ?: return null
+                element.bodyExpression?.statements?.singleOrNull() as? KtCallableReferenceExpression ?: return null
             val callableReference = callableReferenceExpression.callableReference
-            val resolvedCall = callableReference.resolveToCall(BodyResolveMode.FULL)
-            val receiverValue = resolvedCall?.let { it.extensionReceiver ?: it.dispatchReceiver }
-            return (receiverValue?.let { "${it.type}" } ?: "") + "::${callableReference.text}"
+            val receiverExpression = callableReferenceExpression.receiverExpression
+            val receiver = if (receiverExpression == null) {
+                ""
+            } else {
+                val descriptor = receiverExpression.getCallableDescriptor()
+                if (descriptor == null || descriptor is ValueParameterDescriptor)
```

Comment:
This is suspicious place for me. Could you please explain why `ValueParameterDescriptor`, and only this descriptor, should be handled separately here?

Reply:
It is for lambda parameter. If remove ` || descriptor is ValueParameterDescriptor`, following tests fail.
- `defaultParameter.kt`
- `it.kt`
- `parameter.kt`
- `parameter2.kt`

Reply:
It should be not just "some" parameter, but this lambda parameter. See this fixing commit: https://github.com/JetBrains/kotlin/commit/b739db3c3573fb24daf1e97c42746d617dea4692.